mypy/semanal.py

```diff
@@ -5354,7 +5387,8 @@ def analyze_try_stmt(self, s: TryStmt, visitor: NodeVisitor[None]) -> None:
                 type.accept(visitor)
             if var:
                 self.analyze_lvalue(var)
-            handler.accept(visitor)
+            with self.inside_except_star_block_set(s.is_star):
```

Comment:
Another edge case, non-star-except inside star-except:
```python
while True:
    try:
        ...
    except* Exception:
        try:
            ...
        except Exception:
            continue  # missing error
        continue  # E: "continue" not allowed in except* block
```
We should avoid setting the flag to False if it's already True. Something like this should do the trick:
```suggestion
            with self.inside_except_star_block_set(self.inside_except_star_block or s.is_star):
```